src/document_add.c

```diff
@@ -203,7 +205,7 @@ static void replyCallback(RSAddDocumentCtx *aCtx, RedisModuleCtx *ctx, void *unu
     if (aCtx->status.code == QUERY_EDOCNOTADDED) {
       RedisModule_ReplyWithError(ctx, "NOADD");
     } else {
-      RedisModule_ReplyWithError(ctx, QueryError_GetError(&aCtx->status));
+      RedisModule_ReplyWithError(ctx, QueryError_GetUserError(&aCtx->status));
```

Comment:
I still think that it is confusing that we have two APIs to get the same error message. Consider merging them

Reply:
@DvirDukhan FYI.
You originally asked for explicit separate APIs.
Raz thinks they are confusing and thinks we should have one API for both behaviors.
WDYT?